aten/src/ATen/cuda/detail/CUDAHooks.h

```diff
@@ -13,6 +13,11 @@ struct CUDAHooks : public at::CUDAHooksInterface {
   std::unique_ptr<THCState, void(*)(THCState*)> initCUDA() const override;
   Device getDeviceFromPtr(void* data) const override;
   Generator* getDefaultCUDAGenerator(DeviceIndex device_index = -1) const override;
+  c10::Stream getCurrentCUDAStream() const override;
+  void recordCUDAEvent(c10::Event* event, const c10::Stream& stream) const override;
+  void blockCUDAStream(const c10::Event& event, const c10::Stream& stream) const override;
+  void destroyCUDAEvent(c10::Event* event) const override;
+  bool queryCUDAEvent(const c10::Event& event) const override;
```

Comment:
We may not need to change anything for this PR, but if we do want to add support for other APIs in the future, CUDAHooks is not a great place to put the stubs, as there is no promise of API uniformity for hooks across device backends. You'll want something more like DeviceGuardImpl, but for events and streams.

Reply:
I'll update the pattern to be analogous to DeviceGuard(Impl).